go-apps/meep-wais/server/wais.go

```diff
@@ -178,49 +198,76 @@ func Stop() (err error) {
 	return sbi.Stop()
 }
 
-func updateStaInfo(name string, ownMacId string, apMacId string, rssi *int32) {
+func updateStaInfo(name string, ownMacId string, apMacId string, rssi *int32, appNames []string) {
 
 	// Get STA Info from DB, if any
-	var staInfo *StaInfo
-	jsonStaInfo, _ := rc.JSONGetEntry(baseKey+"UE:"+name, ".")
-	if jsonStaInfo != "" {
-		staInfo = convertJsonToStaInfo(jsonStaInfo)
+	var staData *StaData
+	jsonStaData, _ := rc.JSONGetEntry(baseKey+"UE:"+name, ".")
+	if jsonStaData != "" {
+		staData = convertJsonToStaData(jsonStaData)
+	}
+
+	//update data rates
+	sumUl := 0.0
+	sumDl := 0.0
+	for _, appName := range appNames {
+
+		metricsArray, err := metricStore.GetCachedNetworkMetrics("*", appName)
```

Comment:
This should be done in the SBI.

Reply:
this is the function updateStaInfo which is called from the sbi but implemented in the wais.go file.... So i think that is the SBI. can you confirm you agree?

Reply:
Yes, the SBI invokes this function, however the wais now depends on the metric store to retrieve the network metrics. I believe this dependency should be in the SBI package only. The SBI package should be the only one that changes to obtain the necessary data.

Reply:
moved the metricStore code to the sbi